bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java

```diff
@@ -4011,11 +3958,72 @@ private static boolean isValidMapInsertion(BType mapType, BValue value) {
         return false;
     }
 
+    @Deprecated
     private static boolean checkFunctionCast(BValue value, BType lhsType) {
-        if (value.getType().getTag() != TypeTags.FUNCTION_POINTER_TAG) {
+        return checkFunctionCast(value.getType(), lhsType);
+    }
+
+    public static boolean isAssignable(BType sourceType, BType targetType) {
```

Comment:
Is it possible to unify this with checkCast(), and have only one method?
most of of the checks seems very similar in both the methods...